utils/sample_data_generators.py

```diff
@@ -0,0 +1,24 @@
+from decimal import Decimal
+import math
+
+
+def generate_decimal(digits=None, decimal_places=None):
+    if not digits:
+        if not decimal_places:
+            digits = 3
+            decimal_places = 2
```

Comment:
Move it to variables declaration instead of `None`. It would be looks better and function will be shorter

Reply:
You can delete all this places where you check if digits or decimal_places exists. Your function should start like this:
```python
def generate_decimal(digits=3, decimal_places=2):
    assert isinstance(digits, int) and digits > 0
    assert isinstance(decimal_places, int) and decimal_places >= 0 # <- if it can be 0 of course :)
.....
```

Reply:
Here's the thing: I would like this function to always return 2 decimal places if possible (digits > 2 and decimal_places > 3), since it's more convenient for me. At the same time I want the two parameters to work independently (so it's replicating original usage, where I define only one of them).

Here's the reason I can't just move these values in declaration: let's say I put in 3 decimal places, but the default total digits number is now 3... So I guess I need to append the digits value to make it work.

Ok, but now let's say that I need a number with 2 digits (since I'm already supposed to make this whole function somewhat reusable, I guess). So now the default decimal places are 2 and if we apply the same logic as before, I will have to append the digits number.

So the result now is different than intended.

In order to make it most convenient, I need it to go through those multiple ifs.
Or I drop the 2 digits case, since I personally don't need it anywhere at all.
Or I split it again into two functions, if you want simpler code.

Reply:
Done